template/hooks/make/run-make

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# shellcheck source=../git_config_wrapper.sh
+. "$GIT_DIR/hooks/git_config_wrapper.sh"
+
+get_hook_config make target target optional test
+
+if ! make --file="$(git rev-parse --show-toplevel)/Makefile" $target
```

Comment:
Be careful that by doing so, you keep the context of the current directory you are in.

For instance, if one of your target execute the `pwd` command, it will be done in your current directory, not in the root directory.  
For `pwd` this might looks like what you expect but it usually won't be the case for other commands.

You probably want to `cd` to your git "toplevel" and then perform a simple `make $target`.

Reply:
Might be better yeah. I usually try to make my `Makefile` independent from the cwd, but that dependency often creeps back, will do!